app/src/test/kotlin/compiler/intermediate_form/LinearizationTest.kt

```diff
@@ -0,0 +1,437 @@
+package compiler.intermediate_form
+
+import compiler.common.intermediate_form.Covering
+import compiler.common.reference_collections.referenceHashMapOf
+import compiler.intermediate_form.Instruction.UnconditionalJumpInstruction.Jmp
+import kotlin.test.Test
+import kotlin.test.assertContains
+import kotlin.test.assertEquals
+import kotlin.test.assertIs
+import kotlin.test.assertSame
+
+class LinearizationTest {
+    private val unconditional = referenceHashMapOf<Instruction, IFTNode>()
+    private val conditional = referenceHashMapOf<Instruction, Triple<IFTNode, Boolean, String>>()
+
+    private val covering = object : Covering {
+        override fun coverUnconditional(iftNode: IntermediateFormTreeNode): List<Instruction> {
+            val instruction = Instruction.NoOp()
+            unconditional[instruction] = iftNode
+            return listOf(instruction)
+        }
+
+        override fun coverConditional(iftNode: IntermediateFormTreeNode, targetLabel: String, invert: Boolean): List<Instruction> {
+            val instruction = Instruction.NoOp()
+            conditional[instruction] = Triple(iftNode, !invert, targetLabel)
+            return listOf(instruction)
+        }
+    }
+
+    private sealed class AsmablePattern
```

Comment:
Maybe `AsmablePatternTest`?

Reply:
We don't test patterns, but use patterns for testing.